middle_end/flambda/un_anf.ml

```diff
@@ -122,43 +122,66 @@ let make_var_info (clam : Clambda.ulambda) : var_info =
   let rec loop ~depth ~environment_vars : Clambda.ulambda -> unit = function
     (* No underscores in the pattern match, to reduce the chance of failing
        to traverse some subexpression. *)
-    | Uvar var -> add_use t var depth; environment_vars
+    | Uvar var -> add_use t var depth;
+    environment_vars
     | Uconst const ->
       (* The only variables that might occur in [const] are those in constant
          closures---and those are all bound by such closures.  It follows that
          [const] cannot contain any variables that are bound in the current
          scope, so we do not need to count them here.  (The function bodies
          of the closures will be traversed when this function is called from
          [Flambda_to_clambda.to_clambda_closed_set_of_closures].) *)
-      ignore_uconstant const; environment_vars
+      ignore_uconstant const;
+      environment_vars
     | Udirect_apply (label, args, dbg) ->
       ignore_function_label label;
-      List.fold_left (loop ~depth ~environment_vars) args;
-      ignore_debuginfo dbg; environment_vars
+      let environment_vars =
+        List.fold_left
+          (fun environment_vars arg -> loop ~depth ~environment_vars arg)
+          environment_vars
+          args
+      in
+      ignore_debuginfo dbg;
+      environment_vars
     | Ugeneric_apply (func, args, dbg) ->
       loop ~depth ~environment_vars func;
-      List.fold_left (loop ~depth ~environment_vars) args;
-      ignore_debuginfo dbg; environment_vars
+      let environment_vars =
+        List.fold_left
+          (fun environment_vars arg -> loop ~depth ~environment_vars arg)
+          environment_vars
+          args
+      in
+      ignore_debuginfo dbg;
+      environment_vars
     | Uclosure (functions, captured_variables) ->
-      List.fold_left (loop ~depth ~environment_vars) captured_variables;
-      List.iter (fun (
+      let environment_vars =
+        List.fold_left
+          (fun environment_vars captured_variables -> loop ~depth ~environment_vars captured_variables)
```

Comment:
It's nice that you noticed that `arg` would not be a very good name in this case, but I would suggest that you use a shorter name if you can, and make it singular instead of plural (this function describes what computation will be performed on each single element of the list of captured variables, so the argument will be a single captured variable).